File: tests/test_iapws_g704_ngases.c
#include <stdio.h>
#include "iapws_g704.h"


int main(void){

    int value;
    int expected;
    int diff;
    int heavywater;


    printf("***** Test ngases in water *****\n");
    heavywater = 0;
    expected = 14;
    value = iapws_g704_capi_ngases(heavywater);
    diff = value - expected;
    printf("%d/%d/%d\t\n", value, expected, diff);
    if(diff != 0){
        return 1;
    }
    
    printf("***** Test ngases in heavywater *****\n");
    heavywater = 1;
    expected = 7;
    value = iapws_g704_capi_ngases(heavywater);
    diff = value - expected;
    printf("%d/%d/%d\t\n", value, expected, diff);
    if(diff != 0){
        return 1;
    }

}

File: test/test_iapws_g704_ngases.f90
program test_iapws_g704_ngases
    use iso_fortran_env
    use iapws_g704
    implicit none

    integer(int32) :: value
    integer(int32) :: expected
    integer(int32) :: diff
    integer(int32) :: heavywater

    print "(A)", "***** F Test ngases in water *****"
    heavywater = 0;
    expected = 14;
    value = iapws_g704_ngases(heavywater);
    diff = value - expected;
    print "(I0.2, A1, I0.2, A1, I0.2)", value, "/", expected, "/", diff
    if(diff /= 0)then
        stop 1
    endif
    
    print "(A)", "***** F Test ngases in heavywater *****"
    heavywater = 1;
    expected = 7;
    value = iapws_g704_ngases(heavywater);
    diff = value - expected;
    print "(I0.1, A1, I0.1, A1, I0.1)", value, "/", expected, "/", diff
    if(diff /= 0)then
        stop 1
    endif

end program